### PR TITLE
Cleanup routing nodes

### PR DESCRIPTION
**Pull request by idegtiarenko**
This change:
* consolidates test utility and helper 
* removes a method that is only used in tests

### CODE REVIEW

**Review by DaveCTurner — APPROVED** [review]
LGTM, one nit/suggestion

File: test/framework/src/main/java/org/elasticsearch/cluster/routing/RoutingNodesHelper.java [hunk 1]
```diff
@@ -14,13 +14,25 @@
...
 import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
 
 import static org.elasticsearch.common.util.CollectionUtils.iterableAsArrayList;
 
 public final class RoutingNodesHelper {
 
     private RoutingNodesHelper() {}
 
+    public static int numberOfShardsOfType(RoutingNodes nodes, ShardRoutingState state) {
```
Comment (DaveCTurner) [inline test/framework/src/main/java/org/elasticsearch/cluster/routing/RoutingNodesHelper.java:25]: naming nit/suggestion

```suggestion
    public static int numberOfShardsWithState(RoutingNodes nodes, ShardRoutingState state) {
```

Comment (idegtiarenko) [reply test/framework/src/main/java/org/elasticsearch/cluster/routing/RoutingNodesHelper.java:25]: Good point, I moved it as is, but it is better to rename

**Review by arteam — APPROVED** [review]
LGTM

**Comment by idegtiarenko** [comment]
@elasticsearchmachine please run elasticsearch-ci/part-2

**Comment by idegtiarenko** [comment]
@elasticsearchmachine please run elasticsearch-ci/part-2

**Pull request merged**